controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -557,8 +557,28 @@ func (r *EphemeralRunnerReconciler) updateStatusWithRunnerConfig(ctx context.Con
 }
 
 func (r *EphemeralRunnerReconciler) createPod(ctx context.Context, runner *v1alpha1.EphemeralRunner, secret *corev1.Secret, log logr.Logger) (ctrl.Result, error) {
+	var envs []corev1.EnvVar
+	if runner.Spec.Proxy != nil {
+		proxyEnvs, err := runner.Spec.Proxy.EnvVars(func(s string) (*corev1.Secret, error) {
+			var secret corev1.Secret
+			err := r.Client.Get(ctx, types.NamespacedName{
+				Name:      s,
+				Namespace: runner.Namespace,
+			}, &secret)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get secret %s: %w", s, err)
+			}
+			return &secret, nil
+		})
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get proxy env vars: %w", err)
+		}
+
+		envs = append(envs, proxyEnvs...)
```

Comment:
I was thinking the same question about using secret for the env here as well, since the proxy URL may contains a username/password